src/main/java/org/gridsuite/sensitivityanalysis/server/service/nonevacuatedenergy/NonEvacuatedEnergyWorkerService.java

```diff
@@ -0,0 +1,936 @@
+                }
+            });
+        }
+    }
+
+    private double getLimitValueFromFactorAndSensitivityValue(SensitivityValue sensitivityValue,
+                                                              SensitivityFactor factor,
+                                                              MonitoredBranchThreshold monitoredBranchThreshold,
+                                                              MonitoredBranchDetailResult monitoredBranchDetailResult) {
+        double limitValue = Double.NaN;
+        String limitName = null;
+
+        int contingencyIndex = sensitivityValue.getContingencyIndex();
+
+        Branch<?> branch = monitoredBranchThreshold.getBranch();
+        Optional<CurrentLimits> currentLimits1 = branch.getCurrentLimits1();
+        Optional<CurrentLimits> currentLimits2 = branch.getCurrentLimits2();
+
+        // use contingencyIndex (-1 or not) to check istN/Nlimitname or istNm1/Nm1Limitname
+        // limits in iidm are in A, so consider only SensitivityFunctionType BRANCH_CURRENT to return
+        // the limit value multiplied by the input coefficient (nCoeff/nm1Coeff)
+        if (contingencyIndex < 0) { // N
+            if (monitoredBranchThreshold.isIstN()) {
+                limitName = "IST";
+                if (factor.getFunctionType() == SensitivityFunctionType.BRANCH_CURRENT_1) {
+                    limitValue = currentLimits1.map(l -> l.getPermanentLimit() * monitoredBranchThreshold.getNCoeff() / 100).orElse(Double.NaN);
+                } else if (factor.getFunctionType() == SensitivityFunctionType.BRANCH_CURRENT_2) {
+                    limitValue = currentLimits2.map(l -> l.getPermanentLimit() * monitoredBranchThreshold.getNCoeff() / 100).orElse(Double.NaN);
+                }
+            } else if (!StringUtils.isEmpty(monitoredBranchThreshold.getNLimitName())) {
+                limitName = monitoredBranchThreshold.getNLimitName();
+                if (factor.getFunctionType() == SensitivityFunctionType.BRANCH_CURRENT_1) {
+                    Optional<LoadingLimits.TemporaryLimit> limit = currentLimits1.flatMap(currentLimits -> currentLimits.getTemporaryLimits().stream().filter(l -> l.getName().equals(monitoredBranchThreshold.getNLimitName())).findFirst());
+                    limitValue = limit.map(temporaryLimit -> temporaryLimit.getValue() * monitoredBranchThreshold.getNCoeff() / 100).orElse(Double.NaN);
+                } else if (factor.getFunctionType() == SensitivityFunctionType.BRANCH_CURRENT_2) {
+                    Optional<LoadingLimits.TemporaryLimit> limit = currentLimits2.flatMap(currentLimits -> currentLimits.getTemporaryLimits().stream().filter(l -> l.getName().equals(monitoredBranchThreshold.getNLimitName())).findFirst());
+                    limitValue = limit.map(temporaryLimit -> temporaryLimit.getValue() * monitoredBranchThreshold.getNCoeff() / 100).orElse(Double.NaN);
+                }
+            }
+        } else {  // N-1
```

Comment:
Can we refactor this to avoid the duplicate with the next else clause? It looks very similar

Reply:
Done